Print all command

benjmarshall opened the pull request:
Add the 'all' command feature which prints all of the beacon log messages. This has been added to the select case statement. 

Moved log reading code into it's own function so it can be called from wherever needed.

Removed global variable for the log data structure as it's not needed and bringing it in local allows for easier testing.

benjmarshall committed:
Create 'all' command to print all log entries

Added the all command which will print all the log entries. In
order to do this the log loading functionality has been moved to it's
own function.

benjmarshall committed:
Remove log data global variable

No need for log data to be a global variable within the current
application structure. The function 'loadLog' now creates and returns
the log data structure. This way the data can be loaded where needed.

benjmarshall committed:
Add os.exit to the end of the all command

Added os.exit(0) to the end of the all command functionality
within the switch statement. This ensures the program will exit with
a success code after all the log messages have been printed and no
other uninteded actions can occur.

jbydeley commented on main.go:
```diff
@@ -67,6 +61,23 @@ func checkError(e error) {
+	checkError(err)
+
+	// unmarshal json and store it in the pointer to beaconLogData {?}
+	// NOTE: figure out if you can use `checkError` here; don't yet understand golang's idiomatic errors handling.
```
IMO, Go's idiomatic way of handling errors is to handle them as close to the sort as reasonable and populate with the needed information to continue or bail. In this case, instead of `panic(err)` we could use `log.Fatal(err)` or we could have a `log.Println("error occurred while parsing bacon_log.json")` and then follow it up with a `log.Fatal(err)` to get additional information. For speed of development, I'd be alright dealing with error handling a little later since in some ways, the stack trace provided by panic might be nice if something goes wrong.

benjmarshall replied:
Agree, we should decide on a common error handling scheme at some point when the functionality has come together.

teesloane replied:
Good to know. Let's make it a point of the refactor to eventually scan the project for errors and make for one consistent style.

jbydeley commented on main.go:
```diff
@@ -67,6 +61,23 @@ func checkError(e error) {
 
+// Load the beacon log from file
+func loadLog() BeaconLog {
+	beaconLogData := new(BeaconLog)
```
So I've yet to have a need to use `new(T)`. In this case, `var beaconLogData BeaconLog` would make more sense. If you wanted to create a pointer, you could do `beaconLogData := &BeaconLog{}` which would be more common then `new(T)`.

benjmarshall replied:
I think you are correct here. I will push some further changes which remove the explicit creation of a pointer.

benjmarshall committed:
Change beaconLogData definition to be more idiomatic

beaconLogData is now defined in the loadLog function using the
var keyword rather than a call to 'new()'.

teesloane commented on main.go:
```diff
@@ -118,22 +129,13 @@ func main() {
 			fmt.Printf("%+v\n", log)
 			os.Exit(0)
+		case "all":
+			beaconLogData := loadLog()
```
I didn't think to load the file only on the case of `all` whereas before the file was already loading into disk. This makes more sense to me.

teesloane reviewed (approved):
Excellent! Let's ship!


![giphy-downsized-large](https://user-images.githubusercontent.com/12987958/29197491-cdad64d2-7e09-11e7-8c95-5a9ae0af6e5e.gif)
